src/ethereum/EtherscanButton/index.tsx

```diff
@@ -22,21 +22,31 @@ const getNetwork = (network: Network) => {
 
 const EtherscanButton = ({
   className,
-  type,
   value,
   network = 'mainnet',
 }: Props): React.ReactElement => {
-  const etherscanLink = `https://${getNetwork(
-    network
-  )}etherscan.io/${type}/${value}`;
+  const type = value.length > 42 ? 'tx' : 'address';
+
+  const onClick = (event: React.MouseEvent<HTMLAnchorElement>): void => {
+    event?.stopPropagation();
```

Comment:
is it needed to check for `event` existence? I think it shouldn't be an issue to assume it will always exist.